app/services/TotalIncomeTaxService.scala

```diff
@@ -38,89 +38,92 @@ trait TotalIncomeTaxService {
-        payload("basic_rate_income_tax_amount"),
-        payload("higher_rate_income_tax"),
-        payload("higher_rate_income_tax_amount"),
-        payload("additional_rate_income_tax"),
-        payload("additional_rate_income_tax_amount"),
-        payload("ordinary_rate"),
-        payload("ordinary_rate_amount"),
-        payload("upper_rate"),
-        payload("upper_rate_amount"),
-        payload("additional_rate"),
-        payload("additional_rate_amount"),
-        payload("other_adjustments_increasing"),
-        payload("marriage_allowance_received_amount"),
-        payload("other_adjustments_reducing"),
-        scottishTax,
-        payload("total_income_tax"),
-        payload("scottish_income_tax"),
-        savingsTax,
-        atsData.income_tax.flatMap(_.incomeTaxStatus).getOrElse(""),
-        rates("starting_rate_for_savings_rate"),
-        rates("basic_rate_income_tax_rate"),
-        rates("higher_rate_income_tax_rate"),
-        rates("additional_rate_income_tax_rate"),
-        rates("ordinary_rate_tax_rate"),
-        rates("upper_rate_rate"),
-        rates("additional_rate_rate"),
-        scottishRates,
-        savingsRates,
-        taxpayerName("title"),
-        taxpayerName("forename"),
-        taxpayerName("surname")
-      )
-    }
+    val savingsRates = SavingsRates(
+      rates("savings_lower_rate"),
+      rates("savings_higher_rate"),
+      rates("savings_additional_rate")
+    )
+
+    TotalIncomeTax(
+      atsData.taxYear,
+      atsData.utr.getOrElse(""),
+      payload("starting_rate_for_savings"),
+      payload("starting_rate_for_savings_amount"),
+      payload("basic_rate_income_tax"),
+      payload("basic_rate_income_tax_amount"),
+      payload("higher_rate_income_tax"),
+      payload("higher_rate_income_tax_amount"),
+      payload("additional_rate_income_tax"),
+      payload("additional_rate_income_tax_amount"),
+      payload("ordinary_rate"),
+      payload("ordinary_rate_amount"),
+      payload("upper_rate"),
+      payload("upper_rate_amount"),
+      payload("additional_rate"),
+      payload("additional_rate_amount"),
+      payload("other_adjustments_increasing"),
+      payload("marriage_allowance_received_amount"),
+      payload("other_adjustments_reducing"),
+      scottishTax,
+      payload("total_income_tax"),
+      payload("scottish_income_tax"),
+      savingsTax,
+      atsData.income_tax.flatMap(_.incomeTaxStatus).getOrElse(""),
+      rates("starting_rate_for_savings_rate"),
+      rates("basic_rate_income_tax_rate"),
+      rates("higher_rate_income_tax_rate"),
+      rates("additional_rate_income_tax_rate"),
+      rates("ordinary_rate_tax_rate"),
+      rates("upper_rate_rate"),
+      rates("additional_rate_rate"),
+      scottishRates,
+      savingsRates,
+      payload("less_tax_adjustment_previous_year"),
+      payload("tax_underpaid_previous_year"),
+      taxpayerName("title"),
+      taxpayerName("forename"),
+      taxpayerName("surname")
+    )
```

Comment:
This isn't for you to fix while trying to retain SA functionality, but I think this wins for the largest case class I've ever seen...

Reply:
Its going to get bigger when we add the PAYE Scottish  income_tax :) Yeah, once everything is merged this is an ideal candidate for a refactor.